SourceryTests/Models/TypeNameSpec.swift

```diff
@@ -62,6 +62,23 @@ class TypeNameSpec: QuickSpec {
                 it("reports array correctly") {
                     expect(TypeName("[Int]").isArray).to(beTrue())
                     expect(TypeName("[[Int]]").isArray).to(beTrue())
+                    expect(TypeName("[[Int: Int]]").isArray).to(beTrue())
+                }
+            }
+
+            context("given dictionary type") {
+                context("as name") {
+                    it("reports dictionary correctly") {
+                        expect(TypeName("[Int: Int]").isDictionary).to(beTrue())
+                        expect(TypeName("[[Int]: [Int]]").isDictionary).to(beTrue())
+                        expect(TypeName("[Int: [Int: Int]]").isDictionary).to(beTrue())
```

Comment:
I'm not sure if I completely understand the code base at the moment, so let me know if I miss any case here.
But probably you mean that the `TypeName` of array of dictionaries should return for `isArray` true, if so I've added a test case for this and it passes on my local machine:

```swift
expect(TypeName("[[Int: Int]]").isArray).to(beTrue())
```